Unable to import parquet file from S3

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
Import parquet files from S3 w/ tidb-lightning tidb mode
[lightning.log](https://github.com/pingcap/tidb-lightning/files/5480119/lightning.log)


2. What did you expect to see?
Import successfully

3. What did you see instead?
Import failed.
```
[2020/11/03 08:09:01.430 +00:00] [DEBUG] [loader.go:310] ["file route result"] [path=test.warehouse.99.parquet] [schema=test] [table=warehouse] [type=parquet]
[2020/11/03 08:09:01.430 +00:00] [INFO] [lightning.go:255] ["load data source completed"] [takeTime=412.430832ms] []
[2020/11/03 08:09:01.441 +00:00] [INFO] [restore.go:283] ["the whole procedure start"]
[2020/11/03 08:09:01.441 +00:00] [INFO] [tidb.go:278] ["skipping check requirements for tidb backend"]
[2020/11/03 08:09:01.492 +00:00] [INFO] [restore.go:640] ["restore all tables data start"]
[2020/11/03 08:09:01.492 +00:00] [INFO] [restore.go:689] ["restore table start"] [table=`test`.`warehouse`]
[2020/11/03 08:09:01.492 +00:00] [INFO] [restore.go:1451] ["load engines and files start"] [table=`test`.`warehouse`]
[2020/11/03 08:09:01.492 +00:00] [INFO] [restore.go:689] ["restore table start"] [table=`test`.`district`]
[2020/11/03 08:09:01.492 +00:00] [INFO] [restore.go:1451] ["load engines and files start"] [table=`test`.`district`]
[2020/11/03 08:09:01.492 +00:00] [INFO] [restore.go:689] ["restore table start"] [table=`test`.`item`]
[2020/11/03 08:09:01.492 +00:00] [INFO] [restore.go:829] ["reusing engines and files info from checkpoint"] [table=`test`.`item`] [enginesCnt=2] [filesCnt=1]
[2020/11/03 08:09:01.492 +00:00] [INFO] [backend.go:227] ["open engine"] [engineTag=`test`.`item`:-1] [engineUUID=9d7854a8-85b0-5426-8ebf-d2b4b85bf20f]
[2020/11/03 08:09:01.492 +00:00] [INFO] [restore.go:899] ["import whole table start"] [table=`test`.`item`]
[2020/11/03 08:09:01.493 +00:00] [INFO] [restore.go:929] ["restore engine start"] [table=`test`.`item`] [engineNumber=0]
[2020/11/03 08:09:01.493 +00:00] [INFO] [restore.go:1015] ["encode kv data and write start"] [table=`test`.`item`] [engineNumber=0]
[2020/11/03 08:09:01.493 +00:00] [INFO] [backend.go:227] ["open engine"] [engineTag=`test`.`item`:0] [engineUUID=06c1e741-f5ff-587f-8155-2d2c77bbfe73]
[2020/11/03 08:09:01.492 +00:00] [INFO] [restore.go:689] ["restore table start"] [table=`test`.`new_order`]
[2020/11/03 08:09:01.493 +00:00] [INFO] [restore.go:1451] ["load engines and files start"] [table=`test`.`new_order`]
[2020/11/03 08:09:04.498 +00:00] [INFO] [restore.go:1918] ["restore file start"] [table=`test`.`item`] [engineNumber=0] [fileIndex=0] [path=test.item.0.parquet:0]
panic: runtime error: index out of range [0] with length 0

goroutine 880 [running]:
github.com/pingcap/tidb-lightning/lightning/mydump.(*ParquetParser).ReadRow(0xc001d85700, 0xeb5e7b3b, 0x3f36660)
        /data1/cy/src/tidb-lightning/lightning/mydump/parquet_parser.go:158 +0x376
github.com/pingcap/tidb-lightning/lightning/restore.(*chunkRestore).encodeLoop(0xc001e10aa0, 0x2562dc0, 0xc0010285d0, 0xc001e75260, 0xc00102e6c0, 0xc001e752c0, 0x253f300, 0xc001e10b60, 0xc001c2aae0, 0xc0004f8f00, ...)
        /data1/cy/src/tidb-lightning/lightning/restore/restore.go:1828 +0x184
github.com/pingcap/tidb-lightning/lightning/restore.(*chunkRestore).restore(0xc001e10aa0, 0x2562dc0, 0xc0010285d0, 0xc00102e6c0, 0x0, 0xc001060880, 0xc001060640, 0xc0004f8f00, 0x0, 0x0)
        /data1/cy/src/tidb-lightning/lightning/restore/restore.go:1920 +0x790
github.com/pingcap/tidb-lightning/lightning/restore.(*TableRestore).restoreEngine.func1(0xc00104c680, 0xc0004f8f00, 0x2562dc0, 0xc0010285d0, 0xc00102e6c0, 0xc000000000, 0xc001060880, 0xc001060640, 0xc00105c7e0, 0xc00104c058, ...)
        /data1/cy/src/tidb-lightning/lightning/restore/restore.go:1062 +0x175
created by github.com/pingcap/tidb-lightning/lightning/restore.(*TableRestore).restoreEngine
        /data1/cy/src/tidb-lightning/lightning/restore/restore.go:1054 +0x64c
```

4. Versions of the cluster

    - TiDB-Lightning version (run `tidb-lightning -V`):

        ```
        Release Version: v4.0.8-1-g4d97351
        Git Commit Hash: 4d973516e633f0cb9a4ff0f8f942dbe27fe6d3c0
        Git Branch: master
        UTC Build Time: 2020-11-03 07:05:51
        Go Version: go version go1.13.15 linux/amd64
        ```
        
    - TiKV-Importer version (run `tikv-importer -V`)

        ```
        (paste TiKV-Importer version here)
        ```
        
    - TiKV version (run `tikv-server -V`):

        ```
        4.0.7
        ```

    - TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        4.0.7
        ```

    - Other interesting information (system version, hardware config, etc):

        >
        >

5. Operation logs
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs
   
6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `inventory.ini` if deployed by Ansible

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus for TiDB-Lightning if possible

--- a/go.mod1
+++ b/go.mod1
@@ -34,7 +34,7 @@ require (
 	github.com/shurcooL/httpgzip v0.0.0-20190720172056-320755c1c1b0
 	github.com/spaolacci/murmur3 v1.1.0 // indirect
 	github.com/tikv/pd v1.1.0-beta.0.20200910042021-254d1345be09
-	github.com/xitongsys/parquet-go v1.5.4-0.20201010004835-f51647f24120
+	github.com/xitongsys/parquet-go v1.5.5-0.20201110004701-b09c49d6d457
 	github.com/xitongsys/parquet-go-source v0.0.0-20200817004010-026bad9b25d0
 	go.etcd.io/etcd v0.5.0-alpha.5.0.20200824191128-ae9734ed278b // indirect
 	go.uber.org/zap v1.16.0

--- a/go.sum1
+++ b/go.sum1
@@ -796,8 +796,8 @@ github.com/vmihailenco/tagparser v0.1.1/go.mod h1:OeAg3pn3UbLjkWt+rN9oFYB6u/cQgq
 github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2 h1:eY9dn8+vbi4tKz5Qo6v2eYzo7kUS51QINcR5jNpbZS8=
 github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2/go.mod h1:UETIi67q53MR2AWcXfiuqkDkRtnGDLqkBTpCHuJHxtU=
 github.com/xitongsys/parquet-go v1.5.1/go.mod h1:xUxwM8ELydxh4edHGegYq1pA8NnMKDx0K/GyB0o2bww=
-github.com/xitongsys/parquet-go v1.5.4-0.20201010004835-f51647f24120 h1:Vq8/hS6jg6KTWQlEaHvnLZMRKkK6amkAtBMjB7UUrBE=
-github.com/xitongsys/parquet-go v1.5.4-0.20201010004835-f51647f24120/go.mod h1:pheqtXeHQFzxJk45lRQ0UIGIivKnLXvialZSFWs81A8=
+github.com/xitongsys/parquet-go v1.5.5-0.20201110004701-b09c49d6d457 h1:tBbuFCtyJNKT+BFAv6qjvTFpVdy97IYNaBwGUXifIUs=
+github.com/xitongsys/parquet-go v1.5.5-0.20201110004701-b09c49d6d457/go.mod h1:pheqtXeHQFzxJk45lRQ0UIGIivKnLXvialZSFWs81A8=
 github.com/xitongsys/parquet-go-source v0.0.0-20190524061010-2b72cbee77d5 h1:XmN4NA9133N6OvDEAR6TVVhFq5NgetYTyeKl1EMNazs=
 github.com/xitongsys/parquet-go-source v0.0.0-20190524061010-2b72cbee77d5/go.mod h1:xxCx7Wpym/3QCo6JhujJX51dzSXrwmb0oH6FQb39SEA=
 github.com/xitongsys/parquet-go-source v0.0.0-20200817004010-026bad9b25d0 h1:a742S4V5A15F93smuVxA60LQWsrCnN8bKeWDBARU1/k=

